bundles/admin/admin-announcements/view/AnnouncementsListHandler.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { announcementsHelper } from './AnnouncementsHelper';
+import { Message } from 'oskari-ui';
+
+/*
+Handler for admin-announcements forms.
+*/
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='admin-announcements' />;
+
+/**
+* Remove item from a localStorage() array
+* @param {String} name  The localStorage() key
+* @param {String} value The localStorage() value
+*/
+const removeFromLocalStorageArray = (name, value) => {
+    // Get the existing data
+    var existing = localStorage.getItem(name);
+    // If no existing data, create an array
+    // Otherwise, convert the localStorage string to an array
+    existing = existing ? existing.split(',') : [];
+    // Get index of announcement id and remove it from the array
+    const index = existing.indexOf(value.toString());
+    if (index > -1) {
+        existing.splice(index, 1);
+    }
+    // Save back to localStorage
+    localStorage.setItem(name, existing.toString());
+};
+
+class ViewHandler extends StateHandler {
+    constructor () {
+        super();
+        this.announcementsHelper = announcementsHelper();
+        this.newTitle = 'Uusi ilmoitus';
+        this.state = {
+            update: false,
+            announcements: [],
+            title: 'Uusi Ilmoitus',
+            updated: false,
+            active: true,
+            activeKey: []
+
+        };
+    }
+
+    getAdminAnnouncements (callback) {
+        this.announcementsHelper.getAdminAnnouncements(function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.getAdminAnnouncementsFailed'));
+                return;
+            }
+            callback(data);
+        });
+    }
```

Comment:
If you just update the state with the results you can remove that hack from the component that triggers the loading on render and pushes it to state afterwards and probably get rid of that updated-variable in state with this as well: 
```suggestion
    fetchAdminAnnouncements () {
        this.announcementsHelper.getAdminAnnouncements(function (err, data) {
            if (err) {
                Messaging.error(getMessage('messages.getAdminAnnouncementsFailed'));
                return;
            }
            this.updateState({
                announcements: data
            });
        });
    }
```